game/missiongenerator/aircraft/waypoints/seadingress.py

```diff
@@ -58,3 +80,40 @@ def add_tasks(self, waypoint: MovingPoint) -> None:
         # Preemptively use ECM to better avoid getting swatted.
         ecm_option = OptECMUsing(value=OptECMUsing.Values.UseIfDetectedLockByRadar)
         waypoint.tasks.append(ecm_option)
+
+        # For DECOY type flights, setup a waypoint task to skip the target waypoint after
+        # the attack task is complete. This is achieved using a switch waypoint task from the
+        # INGRESS point to the SPLIT point. This tasking prevents the flights continuing to
+        # overfly the target. See link below for the details of this issue
+        # https://github.com/dcs-liberation/dcs_liberation/issues/2781
+        if self.flight.loadout.has_weapon_of_type(WeaponType.DECOY):
+            ingress_index = None
+            split_index = None
+            if isinstance(self.flight.state, InFlight):
+                index = 2
```

Comment:
I think this is a bit too sensitive to the details of what the next waypoint is. Am I correct that the goal here is to determine the index of the next AI waypoint after the IP? And the extra funkiness here is because we need the _DCS_ waypoint index rather than the Liberation index (which can differ because of player-only waypoints and because of mid-flight generation)?

If that's right, I think this can be defended against that (and simplified a great deal as an added bonus) by adding the current index (name it something obvious like `generated_waypoint_idx` or maybe `waypoint_idx_offset_from_mission_start` so we don't confuse the two indexes) as a member of `PydcsWaypointBuilder`. Then I think this block of code can be reduced to something like:

```python
switch_waypoint_task = SwitchWaypoint(self.generated_waypoint_idx, self.generated_waypoint_idx + 1)
```

If you need to handle the error case where the IP is the last waypoint (DCS may not care), you can also pass the list of waypoints to generate to the builder and bounds check on that.

Those two values, btw, I think are just the `idx` and `filtered_points` here: https://github.com/dcs-liberation/dcs_liberation/blob/b6059f692eec3f58ca305a3253ed480a43e301da/game/missiongenerator/aircraft/waypoints/waypointgenerator.py#L112

lmk if I've overlooked something that makes that incorrect :)

Reply:
That's a great point -- this refactoring made this class alot cleaner.